File: maxsubarray/maxsubarray.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int maxsubarray(vector<int> A){
    vector<int> S(A.size(), 0);
    S[0] = A[0];
    
    for (int i = 1; i < A.size(); i++){
        S[i] = max(A[i] + S[i-1], A[i]); 
    }

    return *(max_element(S.begin(), S.end()));
}

int main(int argc, char * argv[]){
    vector<int> A = {-2, 1, -3, 4, -1, 2, 1, -5, 4};
    cout << "Maximum subarray sum: " << maxsubarray(A) << endl;
}

File: maxsubarray/maxsubarray.py
def maxsubarrayH(A, i, solution):
    

def maxsubarray(A):
    
    solution = [A[0]]

    for i in range(1, n):
        solution.append(maxsubarrayH(A, i))
    
    return max(solution)

A = [-2, 1, -3, 4, -1, 2, 1, -5, 4]



print(maxsubarray(A, 0))


